api/src/main/java/io/grpc/ClientInterceptor.java

```diff
@@ -24,10 +24,14 @@
  * <p>Implementers use this mechanism to add cross-cutting behavior to {@link Channel} and
  * stub implementations. Common examples of such behavior include:
  * <ul>
- * <li>Adding credentials to header metadata</li>
  * <li>Logging and monitoring call behavior</li>
+ * <li>Adding metadata for proxies to observe</li>
  * <li>Request and response rewriting</li>
  * </ul>
+ *
+ * <p>Providing authentication credentials is better served by {@link
+ * CallCredentials}. But a {@code ClientInterceptor} could set the {@code
+ * CallCredentials} within the {@link CallOptions}.
```

Comment:
Should we say something about whether this is undesirable or not?

Reply:
ClientInterceptor can work. But using CallCredentials is easier and generally better. It's sort of in-depth, so didn't want to really get into it in this documentation.

We could end up putting some benefits of CallCredentials in the CallCredentials class though.